### ISSUE TITLE
Does SC 2.4.3 Focus Order relate to the predictability of interactions such as static elements?

### ISSUE THREAD

**Issue by anevins12** [w3c/wcag#3871]
Hello!

Is there any part of the SC 2.4.3 Focus Order or "spirit" of it that relates to how predictable interactions are when things receive keyboard focus? 

This is in context of static or disabled controls that receive keyboard focus. 

Some folks say that because static or disabled controls are in the sequential focus order, there is an expectation that they perform an action. That they're interactive. When they're not interactive, the user's perception that these are interactive is changed. Or in other words, it's illogical to them why something non-interactive is focusable and when it's illogical there is meaning loss.

Or does the SC only concern itself with preserving meaning in _content_ or _operation_ in controls either when:
- Meaning inside content changes when navigated sequentially.
- Operation in controls changes when navigated sequentially.

And static content that receives sequential focus doesn't have meaning or operation, therefore there is no meaning or operation preserved. This doesn't take into account how users perceive the interactivity of the sequential navigation order.

I understand that some folks here think that static elements or disabled elements aren't covered in the SC and are set to clarify this in the Understanding doc via a pending PR.

However, I would love to understand a reason for this in the meanwhile.

**Comment by patrickhlauke** [comment]
hard to work out what you're actually asking here without a clear (visual) example. i *think* what you're getting at is: if a control is disabled, so doesn't receive focus, can we fail that under 2.4.3 because the user may not understanding why that control is disabled? if so, that feels like a bit of a reach to me. and what would the suggested fix be? make disabled controls focusable? remove disabled controls visually, completely, so there's no confusion?

**Comment by anevins12** [comment]
Let’s scrap disabled controls from my obscure opening post and go with static elements.

Thank you for trying to decipher this.

Here’s an example: A webpage has headings that have tabindex=0 and the headings are not interactive but do receive keyboard focus.

**Comment by mbgower** [comment]
There are several situations where a non-interactive item may take focus. I am going to distinguish between situations involving  focusable elements and those involving tabbable elements.

Some typically non-interactive items be given a `tabindex="-1"`  --- focus may be placed there by the author, but the element is otherwise not in the tab order. This may be done for a specifc consideration, such as allowing scrolling to take place in regions without interaction components, or to reposition content in the viewport, such as a relative page anchor. 

Other non-actionable controls may be persistently in the tab order. They are reachable by pressing Tab. IMO, these are less common and less desirable, since putting a non-operable control in the tab order is likely to be a source of confusion. Why is this element in the tab order? What am I supposed to do?

In either situation, Focus Order can be used to help ensure the implementation happens in a predictable and beneficial way:
> If a [Web page](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html#dfn-web-page) can be [navigated sequentially](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html#dfn-navigated-sequentially) and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability.

That wording is clearly very relevant to the second scenario involving non-interative tabbable elements. However, I have also applied the second phrase 'components receive focus in an order that preserves.." to assess uses of tabindex="-1".

**Comment by anevins12** [comment]
Hey @mbgower , I’m unsure that the wording is clear enough that it relates to interactions of non-interactive tabbable elements. 

This is because it can be interpreted as though there is no meaning or operation loss when navigating to say static headings. A heading named “Summary of Product” will not lose meaning if it is tabbed to. Similarly, it has no operation therefore there is no operation to preserve.

If the definition of the criterion is ambiguous (which I think it is) there is usually good explanation in the Understanding doc and techniques to give further context. However, there’s nothing there about static elements.

**Comment by patrickhlauke** [comment]
> Other non-actionable controls may be persistently in the tab order. They are reachable by pressing Tab. IMO, these are less common and less desirable, since putting a non-operable control in the tab order is likely to be a source of confusion.

does "less desirable" lead to "fails the SC"? I'd argue that, as ever, it requires a bit of a judgement call. A site that has a few non-interactive things that still fall into focus can be a bit annoying, but I wouldn't hard-fail them. If on the other hand every. Single. Element. Receives. Focus. I'd start to consider it an actual focus order failure.

And we have discussed this previously (and still ongoing) here https://github.com/w3c/wcag/issues/1572 so I'd suggest closing this issue in favour of continuing conversation there?
